:recycle: Change board allocations from heap to stack

ickule opened the pull request.

kaoliiny commented:
If must also be resolved

kaoliiny commented:
No. Please learn about static/dynamic memory allocation.

kaoliiny commented:
Rule is simple: use the stack whenever you can

ickule commented:
I believe that PR is made obsolete by https://github.com/ickule/yagolic/pull/31

kaoliiny commented on include/copy_board.h:
```diff
@@ -1,7 +1,7 @@
 #define COPY_BOARD_H_INCLUDED
 
-void copy_board(int rows, int cols, int *board, int *copy);
+void copy_board(int rows, int cols, int board[rows][cols], int copy[rows][cols]);
```
Does it even work?

kaoliiny replied:
Spoiler: no

ickule replied:
It does actually.
Point taken though, I did not intend to keep using this notation.

kaoliiny commented on include/spread_life.h:
```diff
@@ -6,7 +6,7 @@
 #include "count_neighbours.h"
 
-void spread_life(int *board, int rows, int cols);
+void spread_life(int rows, int cols, int board[rows][cols]);
```
Please Google how to pass arrays into function

kaoliiny reviewed (changes requested):
fix comments

ickule commented on src/count_neighbours.c:
```diff
@@ -1,101 +1,101 @@
-      board[(x + 1) * cols + y] +
-      board[(x + 1) * cols + (y + 1)];
+      board[x][y+1] +
+      board[x+1][y] +
```
Yeah, I figured out it is not really worth it keeping it a 2D matrix as the code got more complex.

ickule commented:
I fixed all the comments and rebased with the latest additions.

Is that what you had in mind?

I did not find a way to not have a fixed size for the board and I could not have the function directly initialised the cells array since their scope are limited to the current function when allocated from stack as far as I understand. 
I am happy to take in suggestions if there is something I missed on how to do that.

As I mentioned in https://github.com/ickule/yagolic/pull/31 I do not like the idea of a fixed board like this and the previous version allocated from heap feels simpler and clearer.
